homeassistant/components/light/mqtt.py

```diff
@@ -563,6 +569,27 @@ def supported_features(self):
             if self._optimistic_brightness:
                 self._brightness = kwargs[ATTR_BRIGHTNESS]
                 should_update = True
+        elif ATTR_BRIGHTNESS in kwargs and \
+                self._topic[CONF_RGB_COMMAND_TOPIC] is not None:
+            rgb = color_util.color_hsv_to_RGB(
+                self._hs[0], self._hs[1], kwargs[ATTR_BRIGHTNESS] / 255 * 100)
+            tpl = self._templates[CONF_RGB_COMMAND_TEMPLATE]
+            if tpl:
+                rgb_color_str = tpl.async_render({
+                    'red': rgb[0],
+                    'green': rgb[1],
+                    'blue': rgb[2],
+                })
+            else:
+                rgb_color_str = '{},{},{}'.format(*rgb)
+
+            mqtt.async_publish(
+                self.hass, self._topic[CONF_RGB_COMMAND_TOPIC],
+                rgb_color_str, self._qos, self._retain)
+
+            if self._optimistic_brightness:
```

Comment:
We also need to make sure `self._optimistic_brightness` is set correctly when operating like this. Currently it's based on the existence of [the brightness state topic](https://github.com/home-assistant/home-assistant/blob/06bb69003745db191a514db39fb4afd20c9d25de/homeassistant/components/light/mqtt.py#L205), but when we're using RGB for brightness control we need to also check if the RGB state topic is defined.

Reply:
I've gotten a bit tied up trying to implement this - is it actually the RGB state topic we need to be checking, or the existence of the RGB command topic / Non-existence of the brightness command topic?

I'm currently thinking we want optimistic brightness if:

```
we are running in optimistic mode
OR
the brightness state topic isn't set and we have a brightness control topic
OR
the rgb state topic isn't set and we don't have a brightness control topic
```

Does this sound correct?

Reply:
I believe this is done as of https://github.com/home-assistant/home-assistant/pull/17075/commits/faabbd4ab701cd7ab33e5e4b21dc26d758a29ca8